atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -235,6 +236,7 @@ public static SerializableTransactionManager create(
             LockServerOptions lockServerOptions,
             boolean allowHiddenTableAccess,
             Class<?> callingClass) {
+        validateRuntimeConfig(runtimeConfig.get());
```

Comment:
We can't push them down to create, because we want to log specifically when:
- the product that uses the atlasdb library calls TM.create() with the new runtime config parameter.
- the runtime config is not present.
This log should target the person managing the stack, to make him/her move parameters from install configs to runtime configs.
We should have incentives other than logging to make products call the new TM.create(), one of them being marking the old calls as deprecated.